### PR TITLE
chore: enable Kubernetes v1.14.0-beta.2

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:

See https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.14.md#changelog-since-v1140-beta1

**Issue Fixed**:

**Requirements**:

- [x] Windows artifacts uploaded to acsmirror blob store
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/common/versions.go [hunk 1]
```diff
@@ -109,6 +109,7 @@ var AllKubernetesSupportedVersions = map[string]bool{
 	"1.14.0-alpha.1": false,
 	"1.14.0-alpha.2": true,
```
Comment (CecileRobertMichon) [inline pkg/api/common/versions.go]: remove alpha.2?

Comment (mboersma) [reply pkg/api/common/versions.go]: I thought in the past we left all alpha + beta versions enabled, then set them all to `false` when there was an official release to replace them. But I see we disabled alpha.1 so I assume we're applying our support policy to prereleases as well? Works for me.

Comment (CecileRobertMichon) [reply pkg/api/common/versions.go]: yeah I'm not sure, I saw that alpha.1 was deprecated which is why I asked. I don't think we're used to having more than 1 pre-release. Either way is fine I think, it doesn't matter much.

**Comment by CecileRobertMichon** [comment]
I reverted #749 to fix unit tests

**Comment by CecileRobertMichon** [comment]
VHD pipeline fix is in #763

**Comment by CecileRobertMichon** [comment]
both 1.14 tests passed in a previous test https://dev.azure.com/aks-engine/AKS-Engine/_build/results?buildId=156 so this can probably merge without waiting for E2E.

**Pull request merged**